controllers/util/util.go

```diff
@@ -54,3 +55,29 @@ func Find[T any](list []T, f func(item *T) bool) *T {
 	}
 	return nil
 }
+
+func AddAnnotation(obj metav1.Object, key string, value string) bool {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+		obj.SetAnnotations(annotations)
+	}
+	if oldValue, exist := annotations[key]; !exist || oldValue != value {
+		annotations[key] = value
+		return true
+	}
+	return false
+}
+
+func AddLabels(obj metav1.Object, key string, value string) bool {
```

Comment:
You are adding a single label
```suggestion
func AddLabel(obj metav1.Object, key string, value string) bool {
```

Reply:
Done